pvlib/iotools/sodapro.py

```diff
@@ -0,0 +1,361 @@
+"""Functions to access data from Copernicus Atmosphere Monitoring Service
+    (CAMS) radiation service.
+.. codeauthor:: Adam R. Jensen<adam-r-j@hotmail.com>
+"""
+
+import pandas as pd
+import requests
+import io
+import warnings
+
+
+CAMS_INTEGRATED_COLUMNS = [
+    'TOA', 'Clear sky GHI', 'Clear sky BHI', 'Clear sky DHI', 'Clear sky BNI',
+    'GHI', 'BHI', 'DHI', 'BNI',
+    'GHI no corr', 'BHI no corr', 'DHI no corr', 'BNI no corr']
+
+# Dictionary mapping CAMS Radiation and McClear variables to pvlib names
+CAMS_VARIABLE_MAP = {
+    'TOA': 'ghi_extra',
+    'Clear sky GHI': 'ghi_clear',
+    'Clear sky BHI': 'bhi_clear',
+    'Clear sky DHI': 'dhi_clear',
+    'Clear sky BNI': 'dni_clear',
+    'GHI': 'ghi',
+    'BHI': 'bhi',
+    'DHI': 'dhi',
+    'BNI': 'dni',
+    'sza': 'solar_zenith',
+}
+
+# Dictionary mapping time steps to CAMS time step format
+TIME_STEPS_MAP = {'1min': 'PT01M', '15min': 'PT15M', '1h': 'PT01H',
+                  '1d': 'P01D', '1M': 'P01M'}
+
+TIME_STEPS_IN_HOURS = {'1min': 1/60, '15min': 15/60, '1h': 1, '1d': 24}
+
+SUMMATION_PERIOD_TO_TIME_STEP = {'0 year 0 month 0 day 0 h 1 min 0 s': '1min',
+                                 '0 year 0 month 0 day 0 h 15 min 0 s': '15min',  # noqa
+                                 '0 year 0 month 0 day 1 h 0 min 0 s': '1h',
+                                 '0 year 0 month 1 day 0 h 0 min 0 s': '1d',
+                                 '0 year 1 month 0 day 0 h 0 min 0 s': '1M'}
+
+
+def get_cams(start_date, end_date, latitude, longitude, email,
+             identifier='mcclear', altitude=None, time_step='1h',
+             time_ref='UT', verbose=False, integrated=False, label=None,
+             map_variables=True, server='www.soda-is.com'):
+    """
+    Retrieve time-series of radiation and/or clear-sky global, beam, and
+    diffuse radiation from CAMS [1]_, [2]_ using the WGET service [3]_.
```

Comment:
Better to avoid references in the first sentence of a docstring -- the first sentence gets copied to `api.rst`, where sphinx sees the `[1]_` and complains that it doesn't know what you're talking about (`/home/docs/checkouts/readthedocs.org/user_builds/pvlib-python/checkouts/1175/docs/sphinx/source/api.rst:490:<autosummary>:1: WARNING: Unknown target name: "1".`).

Reply:
I have corrected this by adding a second sentence.